pipeline/src/lib.rs

```diff
@@ -59,7 +65,7 @@ impl HostContext {
                         as char;
                     match fd {
                         // stdin, stdout and stderr are supported by the default callback
-                        0..=2 => return Err(format!("Unsupported file descriptor: {fd}")),
+                        0 => return Err(format!("Unsupported file descriptor: {fd}")),
```

Comment:
Comment right above needs to be updated i think.
Also looking at the code, who handles fds 1 and 2? is `HostContext` only used for riscv? 
`handle_simple_queries_callback` also seems to handle 1 and 2?

Reply:
IMO there should be no special treatment for 1 and 2 in the callback itself, the host should just read it afterwards from pipeline or session if needed

Reply:
but there is code in `handle_simple_queries_callback` to handle it... is the `HostContext` callback always present? which one handles it?

Reply:
I removed that though, check `pipeline/src/lib.rs` here